hollow/src/main/java/com/netflix/hollow/core/schema/HollowSchema.java

```diff
@@ -47,6 +48,24 @@
  */
 public abstract class HollowSchema {
 
+    /**
+     * A special constant for determining if a hollow set or map schema supports a hash key
+     * for ordinal values.
+     */
+    static final String ORDINAL_HASH_KEY_FIELD_NAME = "0rdinal";
+
+    /**
+     * A special constant for determining if a hollow set or map schema supports a hash key
+     * for ordinal values.
+     */
+    static final String[] ORDINAL_HASH_KEY_FIELD_NAMES = { ORDINAL_HASH_KEY_FIELD_NAME };
```

Comment:
This concept is currently represented as an empty array.  I am not seeing what this change in serialization format of this concept is providing over the current representation?  Was the current representation incomplete in some way?  If so, can we fix it rather than change it?

Reply:
The problem is the varargs constructor:
```
    public HollowSetSchema(String schemaName, String elementType, String... hashKeyFieldPaths)
```
Although a `null` or empty array outside of this constructor can mean different things it does not mean the same thing when passed as an argument since the constructor cannot differentiate between the passing of an argument or not e.g. `new HollowSetSchema("A", "B")` results in the implicit passing of an empty `hashKeyFieldPaths` array.

The string "0rdinal" never appears in serialized form or accessible via a public API.  It's just an internal trick so `HollowSetSchema` can differentiate between null/empty and a hash key for ordinals.

Reply:
FTR i removed this feature.